server/src/com/cloud/api/ApiServlet.java

```diff
@@ -290,6 +292,19 @@ void processRequestInContext(final HttpServletRequest req, final HttpServletResp
                 CallContext.register(accountMgr.getSystemUser(), accountMgr.getSystemAccount());
             }
 
+            if (CallContext.current().getCallingAccount().getType() == Account.ACCOUNT_TYPE_ADMIN) {
+                s_logger.trace("CIDRs from which Admin accounts are allowed to perform API calls: " + adminCidrs);
+                if (!NetUtils.isIpInCidrList(InetAddress.getByName(remoteAddress), adminCidrs.split(","))) {
+                    auditTrailSb.append(" " + HttpServletResponse.SC_UNAUTHORIZED + " " + "IP-Address of remote not in configured Admin CIDR list");
+                    s_logger.warn("Request by accountId " + CallContext.current().getCallingAccount().getId() +
+                                  " was denied since " + remoteAddress + " does not match " + adminCidrs);
+                    final String serializedResponse =
+                            apiServer.getSerializedApiError(HttpServletResponse.SC_UNAUTHORIZED, "IP-Address of remote not in configured Admin CIDR list",
+                                    params, responseType);
+                    HttpUtils.writeHttpResponse(resp, serializedResponse, HttpServletResponse.SC_UNAUTHORIZED, responseType, apiServer.getJSONContentType());
```

Comment:
@wido Great work, works fine! I'd recommend adding a `return;` line after the `writeHttpResponse` on line 304 because otherwise it will generate invalid json:

```
Error processing json: {"listvirtualmachinesresponse":{"uuidList":[],"errorcode":401,"errortext":"IP-Address of remote not in configured Admin CIDR list"}}{"listvirtualmachinesresponse":{}}
```

With `return;` added it works as expected:

![image](https://cloud.githubusercontent.com/assets/1630096/25055499/7e9166d0-2163-11e7-972e-acdef269fee3.png)

Regards, Remi